index.js

```diff
@@ -1,30 +1,65 @@
 const cmd = require('node-cmd')
 const fs = require('fs');
+const { sep: slash } = require('path');
 const path = require('path');
+const loaderUtils = require('loader-utils');
+
+const properName = name => name.replace(/^./, c => c.toUpperCase());
+const listify = array => array.join(', ')
+    // make a comma-separated list ending with a '&' separator
+    .replace(/(, )[^,]*$/, s => ' & ' + s.split(', ')[1]);
 
 module.exports = function (source) {
 
-    var callback = this.async();
+    const compilers = {
+        transcrypt: {
+            switches: '-b -n',
+            folder: `.${slash}__javascript__`,
+            install: 'pip install transcrypt',
+            python_version: '3.x'
+        },
+        jiphy: {
+            switches: '',
+            folder: `.${slash}`,
+            install: 'pip install jiphy',
+            python_version: '2.x'
+        }
+    };
+
+    const options = loaderUtils.getOptions(this);
+    const compilerName = options && options.compiler || 'transcrypt';
+    const compiler = compilers[compilerName];
+
+    if (!compiler) {
+        throw new Error(`py-loader only supports ${
+                listify(Object.keys(compilers).map(properName))
+            } compilers at present. See README.md for information on using it with others.`);
+    }
+
+    compiler.name = compilerName;
+
+    const callback = this.async();
 
-//    console.log(source);
-    var entry = this._module.resource;
+    const entry = this._module.resource;
+    //console.log(`py-loader: compiling ${entry} with ${compilerName}...`);
 
-    var name = path.basename(entry, ".py");
-    console.log("entry " + name)
+    const name = path.basename(entry, ".py");
 
     if (!entry.toLowerCase().endsWith(".py")) {
-        console.log("This loader only handles .py files. This could be a problem with your webpack.config.js file. Please add a rule for .py files in your modules entry.");
+        console.warn("This loader only handles .py files. This could be a problem with your webpack.config.js file. Please add a rule for .py files in your modules entry.");
         callback(null, source);
     }
 
-    cmd.get('transcrypt -b -n ' + entry, function(err, data, stderr) {
+    cmd.get(`${compiler.name} ${compiler.switches} ${name}.py`, function(err, data, stderr) {
         if (!err) {
-            js = fs.readFileSync("./__javascript__/" + name + ".js", "utf8")
+            const filename = `${compiler.folder}${slash}${name}.js`;
+            js = fs.readFileSync(filename, "utf8");
+            fs.unlinkSync(filename);
```

Comment:
@martim00 so on line 57 the output file the compiler generates gets deleted. To reduce the risk of data loss should we make sure the compiler’s output goes into a temporary folder instead?

Reply:
Yeah, I think it could be better. But since it depends on the compiler switch to change the output folder we'll have to expose it in the compiler settings. Do you think it pays the tradeoff to have another option in the settings? Or maybe we can live with that since it is already a tmp generated file? 

Also, sorry for the delay in response. I think I'll have a time to work in this project by the weekend. I'm experimenting with javascript dependencies inside python code (https://github.com/martim00/python-webpack-loader/issues/2). I don't know if jiphy has this problem. What I need is way to call a javascript library inside python code. Like using axios, for example. 

class Foo:
   def call_api(self):
       # using js_get is the way to tell transcrypt to keep 'get' in the generated javascript
       axios.js_get("www.google.com")

If I just use the axios api the webpack does not include it in the python module dependencies and we get a undefined axios. I think it is because we have to specify the es6 import 'axios' at the top of the python file. Or we could use a solution like https://github.com/webpack-contrib/css-loader which transform any css @import syntax into js import/require so the webpack can understand that the css file depends on another and bundle it together.

Do you have any suggestions on that?